inception-project/inception
# #3831 - Better support a direct-access workflow

**What's in the PR**
- Added global setting to determine which project roles should be able to access the project dashboard
- Hide the menubar if users are not allowed to access the project dashboard
- If project dashboard access is disabled, do not redirect user to dashboard e.g. if a document is not accessibly on the annotation page - instead show a respective message on that page
- Added option in the project settings to configure if random access to documents is supported for the matrix workload
- Added trait editors to workload modes

**How to test manually**
* Disable random access to documents and try accessing directly accessible as well as non-accessible documents, try closing them, etc.
* Disable dashboard access e.g. for annotators and see what happens if you still try to access the dashboard as a user having only the annotator role. Also try to see if you get redirected to the dashboard, e.g. when trying to access inaccessible documents or when closing the last document as problematic (which locks it for the annotator), etc.

**Automatic testing**

**Documentation**
